### PR TITLE
feat:  Extract and generalise ledger types

### PR DESCRIPTION
**Pull request by ryanbtp**
* asnc-sawtooth-sdk created containing generic client types used by opa-tp-protocol

References: CHRON-241

### CODE REVIEW

File: crates/async-sawtooth-sdk/src/zmq_client.rs [hunk 1]
```diff
@@ -54,24 +54,52 @@ pub enum SawtoothCommunicationError {
...
+/// A trait representing a communication channel for sending request and receiving
+/// response messages to/from the Sawtooth network.
+///
+/// This trait defines methods for sending a single message and waiting for a response,
+/// listening on the channel for response messages, and reconnecting to the Sawtooth
+/// network if the connection is lost or interrupted.
+///
 #[async_trait::async_trait]
 pub trait RequestResponseSawtoothChannel {
-    // Send a message and wait for a response, decoding the response as a
```
Comment (mtbc) [inline crates/async-sawtooth-sdk/src/zmq_client.rs:59]: Did we mean to lose this line?

Comment (ryanbtp) [reply crates/async-sawtooth-sdk/src/zmq_client.rs:59]: Nope!

File: crates/opactl/src/main.rs [hunk 2]
```diff
@@ -82,7 +91,9 @@ async fn ambient_transactions<
         debug!(waiting_for=?goal_tx_id, max_steps=?max_steps);
         let goal_clone = goal_tx_id.clone();
 
-        let stream = reader.state_updates(None, Some(max_steps)).await;
+        let mut stream = reader
```
Comment (suchapalaver) [inline crates/opactl/src/main.rs:94]: You're getting an unnecessary_mut clippy warning on this definition.
```suggestion
        let stream = reader
```

**Review by suchapalaver — APPROVED** [review]

**Review by suchapalaver — APPROVED** [review]

**Review by suchapalaver — APPROVED** [review]

**Review by suchapalaver — APPROVED** [review]

**Comment by scealiontach** [comment]
@ryanbtp needs a rebase and I think you are good to go here.

**Review by suchapalaver — APPROVED** [review]

**Pull request merged**